### PR TITLE
[IMP] Users: Enforce two-factor authentication

### PR DESCRIPTION
**Pull request by jero-odoo**
Docs task: ﻿[﻿https://www.odoo.com/mail/view?model=project.task&res_id=4032350&access_token=b0dd44e7-855e-4bbb-9c17-1ae9fc856106](https://www.odoo.com/mail/view?model=project.task&res_id=4032350&access_token=b0dd44e7-855e-4bbb-9c17-1ae9fc856106)

Updating 2fa documentation to include a section on Enforcing 2fa for database users.

Additionally, updated images and content for v17

### CODE REVIEW

File: content/applications/general/users/2fa.rst
Comment (hojo-odoo) [inline content/applications/general/users/2fa.rst:9]:
```suggestion
Two-factor authentication (2FA) is a good way to improve security, and to prevent unauthorized persons
from accessing user accounts.
```

Comment (hojo-odoo) [inline content/applications/general/users/2fa.rst:9]:
```suggestion
Two-factor authentication (2FA) is a good way to improve security, and to prevent unauthorized persons
from accessing user accounts.
```

File: content/applications/general/users/2fa.rst
Comment (hojo-odoo) [inline content/applications/general/users/2fa.rst:11]:
```suggestion
Practically, |2fa| means storing a secret inside an *authenticator*, usually on a mobile phone, and
```

File: content/applications/general/users/2fa.rst
Comment (hojo-odoo) [inline content/applications/general/users/2fa.rst:14]:
```suggestion
This means an attacker needs to guess the account password *and* have access to the authenticator,
```

File: content/applications/general/users/2fa.rst
Comment (hojo-odoo) [inline content/applications/general/users/2fa.rst:43]:
```suggestion
After selecting an authenticator, log in to Odoo, then navigate to :menuselection:`My Profile`.
```

**Review by hojo-odoo — APPROVED** [review]
Hey @jero-odoo,

Great job on this update. I'm approving with a few small suggestions.

Thanks,
John

**Comment by jero-odoo** [comment]
Hey @ksc-odoo this is ready for final review. Thanks!

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:9]:
```suggestion
*Two-factor authentication (2FA)* is a way to improve security, and prevent unauthorized
persons from accessing user accounts.
```
suggesting the use of italics here because this sentence reads like a definition...which we typically format with italics surrounding the main term.
Also, opted to remove "good" to avoid any editorializing

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:15]:
```suggestion
This means an unauthorized user would need to guess the account password *and* have access to the authenticator,
which is a more difficult proposition than either one or the other.
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:21]:
```suggestion
   These lists are just examples. They are **not** endorsements of any specific software.
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:38]:
```suggestion
   commonly used. This is **not** an endorsement of the product.
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:40]:
```suggestion
Two-factor authentication setup
```
this suggestion is based on the guidance we were given in the past to avoid using "-ing" words in headers...

ALT: Two-factor authentication configuration

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:43]:
```suggestion
After selecting an authenticator, log in to Odoo, then click the profile avatar in the upper-right corner, and select :guilabel:`My Profile` from the resulting drop-down menu.
```
SUGGESTION: just to add some more specific directions about how to access the My Profile page... 
a supplemental screenshot of this could also be beneficial, but it's totally your call 👍

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:53]:
```suggestion
continue. Enter the appropriate password, then click :guilabel:`Confirm Password. Next, a :guilabel:`Two-Factor Authentication Activation` pop-up window appears, with a |QR|.
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:62]:
```suggestion
   If scanning the screen is not possible (e.g. the setup is being completed on the *same*
```
here's my reasoning behind this suggestion:

'Set up' (two words) is a verb phrase, meaning it describes an action. For example, 'I will set up the computer.' 

'Setup' (one word) is a noun or an adjective, referring to a particular arrangement or configuration. For example, you might say "the setup of the computer is easy" or "this PC's setup is rather odd"

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:64]:
```suggestion
   device as the authenticator application), clicking the provided :guilabel:`Cannot scan it?` link, or copying the secret to
   manually set up the authenticator, is an alternative.
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:85]:
```suggestion
To confirm |2fa| setup is complete, log out of Odoo.
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:96]:
```suggestion
   If a user loses access to their authenticator, an administrator **must** deactivate |2fa| on the
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:112]:
```suggestion
After installation is complete, go to :guilabel:`Settings app: Permissions`. Tick the checkbox labeled,
```

File: content/applications/general/users/2fa.rst
Comment (ksc-odoo) [inline content/applications/general/users/2fa.rst:117]:
```suggestion
   Selecting :guilabel:`All users` applies the setting to portal users, in addition to employees.
```

**Review by ksc-odoo — APPROVED** [review]
hey, @jero-odoo -- just finished my Final Review of this PR. Great work. Just have a small handful of comments/suggestions, that require your attention. Approving now 👍 If you have any questions, don't hesitate to reach out. Other than that, once these necessary changes are implemented, you can feel free to tag this for Tech Review. Thanks! 👏

**Comment by jero-odoo** [comment]
@samueljlieber this is ready for tech review. Thanks!

**Comment by jero-odoo** [comment]
@StraubCreative

File: content/applications/general/users/2fa.rst
Comment (StraubCreative) [inline content/applications/general/users/2fa.rst:15]: Wording at the end is strange?
```suggestion
authenticator, which is a more difficult proposition.
```

File: content/applications/general/users/2fa.rst
Comment (StraubCreative) [inline content/applications/general/users/2fa.rst:33]: Need a line break here the list isn't rendering correctly
```suggestion

- `1Password <https://support.1password.com/one-time-passwords/>`_
```

File: content/applications/general/users/2fa.rst
Comment (StraubCreative) [inline content/applications/general/users/2fa.rst:62]: think this is better as a tip
```suggestion
.. tip::
```

File: content/applications/general/users/2fa.rst
Comment (StraubCreative) [inline content/applications/general/users/2fa.rst:98]: login = noun
log in = verb
```suggestion
   account before the user can log in.
```

**Review by StraubCreative — APPROVED** [review]
Hi @jero-odoo 
Nice, clean doc.
I have a couple of quick fixes, which I'll push up momentarily and merge.
Thanks for your work on this!

**Comment by StraubCreative** [comment]
Tech review feedback applied on[85cbba2](https://github.com/odoo/documentation/pull/10237/commits/85cbba27bc47c9eafd87c6c25425514adecb2c3f)

All checks pass ✅ 
...
@robodoo r+

**Pull request closed without merging**
